src/main/java/carpet/script/api/Scoreboards.java

```diff
@@ -147,5 +154,295 @@ public static void apply(Expression expression)
             scoreboard.setObjectiveSlot(slot, objective);
             return (_c, _t) -> new NumericValue(slot);
         });
+
+        expression.addLazyFunction("team_list", -1, (c, t, lv) ->
+        {
+            if(lv.size() > 1) throw new InternalExpressionException("'team_list' requires zero or one parameters");
+            CarpetContext cc = (CarpetContext)c;
+            ServerScoreboard scoreboard = cc.s.getMinecraftServer().getScoreboard();
+            if(lv.size() == 0) {
+                Value ret = ListValue.wrap(scoreboard.getTeamNames().stream().map(StringValue::new).collect(Collectors.toList()));
```

Comment:
I suggest you use StringValue::of - it handles better null values if encounters one. Going forward that would be the default way of getting these.